cognite/client/_api/datapoint_tasks.py

```diff
@@ -621,16 +635,18 @@ def get_next_payload(self) -> Optional[Dict[str, Any]]:
             return None
         return self._create_payload_item(math.inf if remaining is None else remaining)
 
-    def _create_payload_item(self, remaining_limit: float) -> Dict[str, Any]:
-        return {
-            **self.identifier.as_dict(),
-            "start": self.next_start,
-            "end": self.end,
-            "limit": min(remaining_limit, self.n_dps_left, self.max_query_limit),
-            **self.agg_kwargs,
-        }
+    def _create_payload_item(self, remaining_limit: float) -> CustomDatapoints:
+        return CustomDatapoints(
+            {
+                **self.identifier.as_dict(),  # type: ignore [misc]
+                "start": self.next_start,
+                "end": self.end,
+                "limit": min(remaining_limit, self.n_dps_left, self.max_query_limit),
+                **self.agg_kwargs,
+            }
+        )
 
-    def store_partial_result(self, res: DataPointListItem) -> None:
+    def store_partial_result(self, res: DataPointListItem) -> Optional[List[SplittingFetchSubtask]]:
```

Comment:
Yeah, but you can't (shouldn't) narrow down the return type of a method when overriding it. That breaks liskov substition principle.

Reply:
If you want that, you shouldn't declare it in the superclass.

Reply:
Or wait, brainfart.. Narrowing the rtype should be fine 🤔 Let me look into this one again.

Reply:
Changed the one on L635 back. We can't make it `None` in `SerialFetchSubtask` though, since a subclass of that again needs it to be `Optional[List[SplittingFetchSubtask]]`, and we shouldn't _broaden_ the rtype in a subclass.

Reply:
Pushed a commit.